src/runtime/terminal.go

```diff
@@ -843,9 +840,19 @@ func Base(env Environment, path string) string {
 	return path
 }
 
+func ReplaceTildePrefixWithHomeDir(env Environment, path string) string {
+	if !strings.HasPrefix(path, "~") {
+		return path
+	}
+	rem := path[1:]
+	if len(rem) == 0 || IsPathSeparator(env, rem[0]) {
+		return env.Home() + rem
+	}
+	return path
```

Comment:
This will return a non-working path when the path for example is `~oops/oh-my-posh.omp.json`. Don the checks above even matter in that case?

Reply:
That's the expected behavior. The validity of a path does not bother the check. Here we only care about the correct replacement for a tilde prefix.

Reply:
I understand, I just want to avoid useless CPU cycles. In this case the eventual result is identical with or without the check. That's why I'd remove it entirely.

Reply:
No matter what path is given, there is no useless time cost for the check that should be guaranteed in the tilde expansion, IMO.